src/platform/imx8/lib/dai.c

```diff
@@ -19,6 +19,18 @@ static struct dai esai[] = {
 	.index = 0,
 	.plat_data = {
 		.base = ESAI_BASE,
+		.fifo[SOF_IPC_STREAM_PLAYBACK] = {
+			.offset		= ESAI_BASE + REG_ESAI_ETDR,
+			.depth		= 96,  /* in 4 bytes words */
```

Comment:
This value 96 might be probably a temporary patch. Eventually we should at some point set it to the real ESAI's FIFO size, ie 128. This would then let more room for the EDMA driver computing the number of bytes to transfer at each major loop iteration.

Reply:
Won't the issue with preloading/potential for channel swap also happen if I gave it the full period size and 0 for watermark? I mean between the emptying of the FIFO and the actual DMA transfer occurring maybe it would underrun and not have one sample to send.